core/test-runtime/src/system.rs

```diff
@@ -113,6 +113,12 @@ fn execute_block_with_state_root_handler(
 
 	let o_new_authorities = <NewAuthorities>::take();
 
+	// symetric removal of finalize for roots to match not putting parent hash as it
+	// seems to touch the state (required key value).
+	<Number>::kill();
+	<StorageDigest>::kill();
+	<NewAuthorities>::kill();
```

Comment:
why is this now needed ?

Reply:
I think I make a mistake in my fix for child (I needed those when I put them in a child trie), there is definitely something going on here that I do not like, but those lines does not make much sense (it seems there is some where root is calculated only from call to this method so initialize may not be call here and thing should be fine as long as those value are not use), I am removing it.